coalib/bearlib/abstractions/Linter.py

```diff
@@ -0,0 +1,680 @@
+                    The config-file-contents as a string or ``None``.
+                """
+                return None
+
+            @staticmethod
+            def create_arguments(filename, file, config_file):
+                """
+                Creates the arguments for the linter.
+
+                You can provide additional keyword arguments and defaults.
+                These will be interpreted as required settings that need to be
+                provided through a coafile-section.
+
+                :param filename:
+                    The name of the file the linter-tool shall process.
+                :param file:
+                    The contents of the file.
+                :param config_file:
+                    The path of the config-file if used. ``None`` if unused.
+                :return:
+                    A sequence of arguments to feed the linter-tool with.
+                """
+                raise NotImplementedError
+
+            @staticmethod
+            def get_executable():
+                """
+                Returns the executable of this class.
+
+                :return:
+                    The executable name.
+                """
+                return options["executable"]
+
+            @classmethod
+            def check_prerequisites(cls):
+                """
+                Checks whether the linter-tool the bear uses is operational.
+
+                :return:
+                    True if available, otherwise a string containing more info.
+                """
+                if shutil.which(cls.get_executable()) is None:
+                    return repr(cls.get_executable()) + " is not installed."
+                else:
+                    if options["prerequisite_check_command"]:
+                        try:
+                            check_call(options["prerequisite_check_command"],
+                                       stdout=DEVNULL,
+                                       stderr=DEVNULL)
+                            return True
+                        except (OSError, CalledProcessError):
+                            return options["prerequisite_check_fail_message"]
+                    return True
+
+            @classmethod
+            def _get_create_arguments_metadata(cls):
+                return FunctionMetadata.from_function(
+                    cls.create_arguments,
+                    omit={"filename", "file", "config_file"})
+
+            @classmethod
+            def _get_generate_config_metadata(cls):
+                return FunctionMetadata.from_function(
+                    cls.generate_config,
+                    omit={"filename", "file"})
+
+            @classmethod
+            def _get_process_output_metadata(cls):
+                return FunctionMetadata.from_function(
+                    cls.process_output,
+                    omit={"self", "output", "filename", "file"})
+
+            @classmethod
+            def get_non_optional_settings(cls):
```

Comment:
yes we need it, it's used by our bear infrastructure to get information on which settings need to be provided
See `Bear.py`